app/scripts/modules/core/src/task/TaskProgressBar.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { OverlayTrigger, Tooltip } from 'react-bootstrap';
+
+import { ITask } from 'core/domain';
+import { robotToHuman } from 'core/presentation/robotToHumanFilter/robotToHuman.filter';
+
+export interface ITaskProgressBarProps {
+  task: ITask;
+}
+
+export function TaskProgressBar(props: ITaskProgressBarProps) {
+  const {
+    task: { steps, id },
+    task,
+  } = props;
+  let tooltip;
+
+  if (task.isRunning) {
+    const currentStep = steps.find(step => step.hasNotStarted || step.isRunning);
+    if (currentStep) {
+      const currentStepIndex = steps.indexOf(currentStep) + 1;
+      tooltip = (
+        <Tooltip id={id}>{`Step ${currentStepIndex} of ${steps.length}: ${robotToHuman(currentStep.name)}`}</Tooltip>
+      );
+    }
+  }
+
+  if (task.isFailed) {
+    const failedStep = steps.find(step => step.isFailed || step.isSuspended);
+    if (failedStep && task.failureMessage) {
+      const failedStepIndex = steps.indexOf(failedStep) + 1;
+      const ellipses = String.fromCharCode(8230);
+      const clipped =
+        task.failureMessage.length > 400 ? task.failureMessage.substring(0, 400) + ellipses : task.failureMessage;
+      tooltip = (
+        <Tooltip id={id}>
+          {`Failed on Step ${failedStepIndex} of ${steps.length}:`}
+          <br />
+          {robotToHuman(failedStep.name)}
+          <br />
+          <br />
+          <strong>Exception:</strong>
+          <p>{clipped}</p>
+        </Tooltip>
+      );
+    } else {
+      tooltip = <Tooltip id={id}>Task failed; sorry, no reason provided</Tooltip>;
+    }
+  }
+
+  const stepsComplete = steps.filter(step => step.isCompleted);
+  const progressBarClass =
+    task.isRunning || task.hasNotStarted
```

Comment:
are these states all mutually exclusive? If so, might be nicer to just use `classnames` and give it an object with all of these.

Reply:
Love it, thanks.